### ISSUE TITLE
lint: add type-guard checker

### ISSUE THREAD

**Issue by Quasilyte** [PieselBois/kfulint#10]
Finds type switch that may benefit from "type switch guard".

So, this code will trigger a warning:
```go
	var x interface{} = 10

	switch x.(type) {
	case int:
		var _ int = x.(int)
	case float32:
		var _ float32 = x.(float32)
	}
```
And the alternative we are proposing:
```go
	var x interface{} = 10

	switch x := x.(type) {
	case int:
		var _ int = x
	case float32:
		var _ float32 = x
	}
```
Another example:
```go
	var v interface{} = point{1, 2}
	switch v.(type) {
	case int:
		return v.(int)
	case point:
		return v.(point).x + v.(point).y
	default:
		return 0
	}
// =>
	switch v := v.(type) {
	case int:
		return v
	case point:
		return v.x + v.y
	default:
		return 0
	}
```

**Comment by Quasilyte** [comment]
This issue should not be closed until:
1) Tests are integrated into `go test` (see #8).
2) Test run is done over big codebases to ensure there is no false positives.

**Comment by Quasilyte** [comment]
I've tried it on many Go stdlib packages and found no false positives.
Tests are also already merged, so this issue can be closed.
